src/parser/statement/copy_statement.cpp

```diff
@@ -11,6 +11,101 @@ CopyStatement::CopyStatement(const CopyStatement &other) : SQLStatement(other),
 	}
 }
 
+string ConvertOptionValueToString(const Value &val) {
+	auto type = val.type().id();
+	switch (type) {
+	case LogicalTypeId::VARCHAR:
+		return KeywordHelper::WriteOptionallyQuoted(val.ToString());
+	default:
+		return val.ToString();
+	}
+}
+
+string CopyStatement::CopyOptionsToString(const unordered_map<string, vector<Value>> &options) const {
+	if (options.empty()) {
+		return string();
+	}
+	string result;
+
+	result += " (";
+	for (auto it = options.begin(); it != options.end(); it++) {
+		if (it != options.begin()) {
+			result += ", ";
+		}
+		auto &name = it->first;
+		auto &values = it->second;
+
+		result += name + " ";
+		D_ASSERT(!values.empty());
+		if (values.size() == 1) {
+			result += ConvertOptionValueToString(values[0]);
+		} else {
+			result += "( ";
+			for (idx_t i = 0; i < values.size(); i++) {
+				auto &value = values[i];
+				if (i) {
+					result += ", ";
+				}
+				result += KeywordHelper::WriteOptionallyQuoted(value.ToString());
+			}
+			result += " )";
+		}
+	}
+	result += " )";
+	return result;
+}
+
+// COPY table-name (c1, c2, ..)
+string TablePart(const CopyInfo &info) {
+	string result;
+
+	if (!info.catalog.empty()) {
+		result += KeywordHelper::WriteOptionallyQuoted(info.catalog) + ".";
+	}
+	if (!info.schema.empty()) {
+		result += KeywordHelper::WriteOptionallyQuoted(info.schema) + ".";
+	}
+	D_ASSERT(!info.table.empty());
+	result += KeywordHelper::WriteOptionallyQuoted(info.table);
+
+	// (c1, c2, ..)
+	if (!info.select_list.empty()) {
+		result += " (";
+		for (idx_t i = 0; i < info.select_list.size(); i++) {
+			if (i > 0) {
+				result += ", ";
+			}
+			result += KeywordHelper::WriteOptionallyQuoted(info.select_list[i]);
+		}
+		result += " )";
+	}
+	return result;
+}
+
+string CopyStatement::ToString() const {
+	string result;
```

Comment:
We have verification for `ToString` round-trips for INSERT/DELETE/SELECT statements in `src/main/client_context.cpp:670` (in `PendingStatementOrPreparedStatementInternal`). Perhaps add the COPY statement there as well?